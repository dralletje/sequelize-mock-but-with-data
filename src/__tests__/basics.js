let Sequelize = require("sequelize");

// I should move to observatory or something,
// but this works for now
const DATE_TO_USE = new Date("2016");
const _Date = Date;
global.Date = jest.fn(() => DATE_TO_USE);
global.Date.UTC = _Date.UTC;
global.Date.parse = _Date.parse;
global.Date.now = _Date.now;

let define = (options = {}) => {
  let sequelize = new Sequelize();
  let User = sequelize.define(
    "User",
    {
      id: {
        type: Sequelize.UUID,
        defaultValue: Sequelize.UUIDV4,
      },
      name: Sequelize.STRING,
      email: Sequelize.STRING,
      age: Sequelize.INTEGER,
      verified: Sequelize.BOOLEAN,
    },
    options
  );
  return { sequelize, User };
};

let some_users_setup = async ({ User }) => {
  await User.create({
    name: "Michiel Dral",
    email: "m.c.dral@gmail.com",
    age: 22,
    verified: true,
  });
  await User.create({
    name: "Ola Beige",
    email: "o.l.beige@gmail.com",
    age: 22,
    verified: true,
  });
  await User.create({
    name: "Jake Strange",
    email: "j.strange@gmail.com",
    age: 24,
    verified: true,
  });
};

it("should work with simple insert", async () => {
  let { sequelize, User } = define();

  let inserted = await User.create({
    name: "Michiel Dral",
    email: "m.c.dral@gmail.com",
    age: 22,
    verified: true,
  });

  expect(inserted).toMatchSnapshot();
});

it("should update a specific user", async () => {
  let { sequelize, User } = define();
  await some_users_setup({ User });

  let jake = await User.findOne({ where: { email: "j.strange@gmail.com" } });
  await User.update(
    {
      age: 25,
    },
    {
      where: { id: jake.id },
    }
  );

  let users = await User.findAll();
  expect(users).toMatchSnapshot();
});

it("should remove a specific user", async () => {
  let { sequelize, User } = define();
  await some_users_setup({ User });

  let jake = await User.findOne({ where: { email: "j.strange@gmail.com" } });
  await User.destroy({
    where: { id: jake.id },
  });

  let users = await User.findAll();
  expect(users).toMatchSnapshot();
});

it("should update a selection of users", async () => {
  let { sequelize, User } = define();
  await some_users_setup({ User });

  await User.update(
    {
      age: 23,
    },
    {
      where: { age: 22 },
    }
  );

  let users = await User.findAll();
  expect(users).toMatchSnapshot();
});

it("should remove a selection of users", async () => {
  let { sequelize, User } = define();
  await some_users_setup({ User });

  await User.destroy({
    where: { age: 22 },
  });

  let users = await User.findAll();
  expect(users).toMatchSnapshot();
});

it("sync will do nothing", async () => {
  let { sequelize, User } = define();
  await some_users_setup({ User });

  let pre_force = await User.findAll();
  await sequelize.sync();
  await User.sync();
  let users = await User.findAll();
  expect(users).toMatchSnapshot(pre_force);
});

it("sequelize.sync({ force: true }) will clear database", async () => {
  let { sequelize, User } = define();
  await some_users_setup({ User });

  await sequelize.sync({ force: true });
  let users = await User.findAll();
  expect(users).toMatchSnapshot([]);
});

it("User.sync({ force: true }) will clear database", async () => {
  let { sequelize, User } = define();
  await some_users_setup({ User });

  await User.sync({ force: true });
  let users = await User.findAll();
  expect(users).toMatchSnapshot([]);
});

it("should not add createdAt and updatedAt if timestamps are disabled", async () => {
  let { sequelize, User } = define({ timestamps: false });
  await some_users_setup({ User });
  let users = await User.findAll();

  expect(users).toMatchSnapshot();
});

it("should only return attributes we ask for", async () => {
  let { sequelize, User } = define({ timestamps: false });
  await some_users_setup({ User });
  let users = await User.findAll({
    attributes: ["name", "age"],
  });

  expect(users).toMatchSnapshot();
});
