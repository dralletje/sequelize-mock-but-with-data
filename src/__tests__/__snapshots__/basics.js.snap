// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User.sync({ force: true }) will clear database 1`] = `Object {}`;

exports[`sequelize.sync({ force: true }) will clear database 1`] = `Object {}`;

exports[`should not add createdAt and updatedAt if timestamps are disabled 1`] = `
Array [
  Object {
    "__table_name__": "User",
    "age": 22,
    "email": "m.c.dral@gmail.com",
    "id": "1",
    "name": "Michiel Dral",
    "verified": true,
  },
  Object {
    "__table_name__": "User",
    "age": 22,
    "email": "o.l.beige@gmail.com",
    "id": "2",
    "name": "Ola Beige",
    "verified": true,
  },
  Object {
    "__table_name__": "User",
    "age": 24,
    "email": "j.strange@gmail.com",
    "id": "3",
    "name": "Jake Strange",
    "verified": true,
  },
]
`;

exports[`should only return attributes we ask for 1`] = `
Array [
  Object {
    "__table_name__": "User",
    "age": 22,
    "name": "Michiel Dral",
  },
  Object {
    "__table_name__": "User",
    "age": 22,
    "name": "Ola Beige",
  },
  Object {
    "__table_name__": "User",
    "age": 24,
    "name": "Jake Strange",
  },
]
`;

exports[`should remove a selection of users 1`] = `
Array [
  Object {
    "__table_name__": "User",
    "age": 24,
    "createdAt": 2016-01-01T00:00:00.000Z,
    "email": "j.strange@gmail.com",
    "id": "3",
    "name": "Jake Strange",
    "updatedAt": 2016-01-01T00:00:00.000Z,
    "verified": true,
  },
]
`;

exports[`should remove a specific user 1`] = `
Array [
  Object {
    "__table_name__": "User",
    "age": 22,
    "createdAt": 2016-01-01T00:00:00.000Z,
    "email": "m.c.dral@gmail.com",
    "id": "1",
    "name": "Michiel Dral",
    "updatedAt": 2016-01-01T00:00:00.000Z,
    "verified": true,
  },
  Object {
    "__table_name__": "User",
    "age": 22,
    "createdAt": 2016-01-01T00:00:00.000Z,
    "email": "o.l.beige@gmail.com",
    "id": "2",
    "name": "Ola Beige",
    "updatedAt": 2016-01-01T00:00:00.000Z,
    "verified": true,
  },
]
`;

exports[`should update a selection of users 1`] = `
Array [
  Object {
    "__table_name__": "User",
    "age": 23,
    "createdAt": 2016-01-01T00:00:00.000Z,
    "email": "m.c.dral@gmail.com",
    "id": "1",
    "name": "Michiel Dral",
    "updatedAt": 2016-01-01T00:00:00.000Z,
    "verified": true,
  },
  Object {
    "__table_name__": "User",
    "age": 23,
    "createdAt": 2016-01-01T00:00:00.000Z,
    "email": "o.l.beige@gmail.com",
    "id": "2",
    "name": "Ola Beige",
    "updatedAt": 2016-01-01T00:00:00.000Z,
    "verified": true,
  },
  Object {
    "__table_name__": "User",
    "age": 24,
    "createdAt": 2016-01-01T00:00:00.000Z,
    "email": "j.strange@gmail.com",
    "id": "3",
    "name": "Jake Strange",
    "updatedAt": 2016-01-01T00:00:00.000Z,
    "verified": true,
  },
]
`;

exports[`should update a specific user 1`] = `
Array [
  Object {
    "__table_name__": "User",
    "age": 22,
    "createdAt": 2016-01-01T00:00:00.000Z,
    "email": "m.c.dral@gmail.com",
    "id": "1",
    "name": "Michiel Dral",
    "updatedAt": 2016-01-01T00:00:00.000Z,
    "verified": true,
  },
  Object {
    "__table_name__": "User",
    "age": 22,
    "createdAt": 2016-01-01T00:00:00.000Z,
    "email": "o.l.beige@gmail.com",
    "id": "2",
    "name": "Ola Beige",
    "updatedAt": 2016-01-01T00:00:00.000Z,
    "verified": true,
  },
  Object {
    "__table_name__": "User",
    "age": 25,
    "createdAt": 2016-01-01T00:00:00.000Z,
    "email": "j.strange@gmail.com",
    "id": "3",
    "name": "Jake Strange",
    "updatedAt": 2016-01-01T00:00:00.000Z,
    "verified": true,
  },
]
`;

exports[`should work with notIn 1`] = `
Array [
  Object {
    "__table_name__": "User",
    "age": 24,
    "email": "j.strange@gmail.com",
    "id": "3",
    "name": "Jake Strange",
    "verified": true,
  },
]
`;

exports[`should work with simple insert 1`] = `
Object {
  "__table_name__": "User",
  "age": 22,
  "createdAt": 2016-01-01T00:00:00.000Z,
  "email": "m.c.dral@gmail.com",
  "id": "1",
  "name": "Michiel Dral",
  "updatedAt": 2016-01-01T00:00:00.000Z,
  "verified": true,
}
`;

exports[`sync will do nothing 1`] = `
Object {
  "0": Object {
    "__table_name__": "User",
    "age": 22,
    "createdAt": 2016-01-01T00:00:00.000Z,
    "email": "m.c.dral@gmail.com",
    "id": "1",
    "name": "Michiel Dral",
    "updatedAt": 2016-01-01T00:00:00.000Z,
    "verified": true,
  },
  "1": Object {
    "__table_name__": "User",
    "age": 22,
    "createdAt": 2016-01-01T00:00:00.000Z,
    "email": "o.l.beige@gmail.com",
    "id": "2",
    "name": "Ola Beige",
    "updatedAt": 2016-01-01T00:00:00.000Z,
    "verified": true,
  },
  "2": Object {
    "__table_name__": "User",
    "age": 24,
    "createdAt": 2016-01-01T00:00:00.000Z,
    "email": "j.strange@gmail.com",
    "id": "3",
    "name": "Jake Strange",
    "updatedAt": 2016-01-01T00:00:00.000Z,
    "verified": true,
  },
}
`;
